dftimewolf/lib/state.py

```diff
@@ -89,90 +106,80 @@ def GetContainers(self, container_class):
     with self._store_lock:
       return self.store.get(container_class.CONTAINER_TYPE, [])
 
-  def SetupModules(self, args):
+  def _SetupModuleThread(self, module_definition):
+    """Calls the module's SetUp() function and sets a threading event for it.
+
+    Callback for _InvokeModulesInThreads.
+
+    Args:
+      module_definition (dict[str, str]): recipe module definition.
+    """
+    module_name = module_definition['name']
+
+    new_args = utils.ImportArgsFromDict(
+        module_definition['args'], self._command_line_options, self.config)
+    module = self._module_pool[module_name]
+
+    try:
+      module.SetUp(**new_args)
+    except Exception as exception:  # pylint: disable=broad-except
+      self.AddError(
+          'An unknown error occurred: {0!s}\nFull traceback:\n{1:s}'.format(
+              exception, traceback.format_exc()),
+          critical=True)
+
+    self._threading_event_per_module[module_name] = threading.Event()
+    self.CleanUp()
+
+  def SetupModules(self, command_line_options):
     """Performs setup tasks for each module in the module pool.
 
     Threads declared modules' SetUp() functions. Takes CLI arguments into
     account when replacing recipe parameters for each module.
 
     Args:
-      args (list[str]): Command line arguments that will be used to replace
-          the parameters declared in the recipe.
+      command_line_options (argparse.Namespace): Command line options that
+          will be used to replace the parameters declared in the recipe.
     """
+    # Note that vars() copies the values of argparse.Namespace to a dict.
+    self._command_line_options = vars(command_line_options)
+    self._InvokeModulesInThreads(self._SetupModuleThread)
 
-    def _SetupModuleThread(module_description):
-      """Calls the module's SetUp() function and sets an Event object for it.
-
-      Args:
-        module_description (dict): Corresponding recipe module description.
-      """
-      new_args = utils.ImportArgsFromDict(
-          module_description['args'], vars(args), self.config)
-      module = self._module_pool[module_description['name']]
-      try:
-        module.SetUp(**new_args)
-      except Exception as exception:  # pylint: disable=broad-except
-        self.AddError(
-            'An unknown error occurred: {0!s}\nFull traceback:\n{1:s}'.format(
-                exception, traceback.format_exc()),
-            critical=True)
-
-      self.events[module_description['name']] = threading.Event()
-      self.CleanUp()
+  def _RunModuleThread(module_description):
```

Comment:
I guess you meant module_definition here? There are a bunch of other misnamed variables that the linter caught, see [travis run](https://travis-ci.org/log2timeline/dftimewolf/jobs/564932330) for details

Reply:
ack, likely introduced due to merge issues.